### ISSUE TITLE
Stack and Detail name case issues in CF 

### ISSUE THREAD

**Issue by jmelchio** [spinnaker/spinnaker#3994]
### Issue Summary:
The case `upper vs. lower` of the name of the server group has the potential to give problems when creating server groups using the CF clouddriver.

### Cloud Provider(s):
CF

### Environment:
Any

### Feature Area (if this issue is UI/UX related, please tag @spinnaker/ui-ux-team):
Clouddriver

### Description:
When trying to create server groups that have the same name when case is **not** considered ie. `app-stack-detail-v000` vs. `app-STACK-detail-v000` in the same ORG and SPACE, Spinnaker will not have a problem passing validation on the front-end but the CF back-end will refuse to create what it considers to be a duplicate app name in an ORG and SPACE.

### Steps to Reproduce:
Create applications described in description and observe errors.

### Additional Details:
NA

**Comment by jmelchio** [comment]
@spinnakerbot add-label provider/cf
@spinnakerbot add-label bug

**Comment by jmelchio** [comment]
We have discovered that this comes down to issue of the CF installations being `case insensitive` while the cache on Spinnaker is `case sensitive`. When users produce pipelines outside of Spinnaker they sometimes mix up the `case` of the names of `ORG` and `SPACE` and when this happens there are issues with parts of the application that rely on cache lookups that refer to `region` which is an amalgam of a `ORG` and `SPACE` name. 

The cache lookups are `case sensitive` and therefor fail when the case of the `ORG` and/or `SPACE` name does not match which produces the issues users are reporting.

We've completed a fix where we do an extra check for this and provide meaningful error messaging to the user so that they will be able to fix their pipelines.
